.github/workflows/security-alerts.yml

```diff
@@ -46,17 +46,43 @@ jobs:
         with:
           python-version: "3.11"
 
-      - name: Install Python dependencies
-        run: |
-          pip install --upgrade pip
-          pip install --no-cache-dir -r requirements.txt
+      - name: Load cached Poetry installation
+        id: cached-poetry
+        uses: actions/cache@v4
+        with:
+          path: ~/.local
+          key: poetry-1.7.1-0
+
+      - name: Install Poetry
+        if: steps.cached-poetry.outputs.cache-hit != 'true'
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.7.1
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v4
+        with:
+          path: .venv
+          key: pydeps-${{ hashFiles('**/poetry.lock') }}
+
+      - name: Install dependencies
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: poetry install --no-interaction --no-root
+
+      - name: Install project
+        run: poetry install --no-interaction
 
       - name: Filter advisories
         id: filter_advisories
         env:        
           REPO: datahub
           REPO_OWNER: datahub-project
-        run: python scripts/filter_advisories.py \
+        run: |
```

Comment:
Any reason we can't do `poetry run python...` here? That way you don't need a separate line for managing the virtualenv

Reply:
I went with the default pattern suggested within the [snok/install-poetry workflow readme](https://github.com/snok/install-poetry?tab=readme-ov-file#testing), primarily to remove a further dependency on poetry.